go/vt/topo/memorytopo/memorytopo.go

```diff
@@ -140,12 +147,8 @@ func (c *Conn) dial(ctx context.Context) error {
 }
 
 // Close is part of the topo.Conn interface.
-// It nils out factory, so any subsequent call will panic.
 func (c *Conn) Close() {
-	c.factory.Lock()
-	f := c.factory
-	defer f.Unlock()
-	c.factory = nil
+	c.closed = true
```

Comment:
Can topo.Conns be used concurrently? If so, you could use an atomic increment and then load to check if closed. Otherwise this could also race reads/writes.

Reply:
@mdlayher They're not intended to be used concurrently here. The main race issue is between these and the watchers which spawn a separate goroutine.